### ISSUE TITLE
PTC (and google) login failing when attempting to re-authenticate

### ISSUE THREAD

**Issue by Krys1s** [AHAAAAAAA/PokemonGo-Map#1315]
I notice this happening when re-authenticating an active instance. If I close out and re-launch it authenticates correctly the first try. Until I close out, it sits in a loop with these errors indefinitely.

`2016-07-22 11:50:38,694 [     pgoapi] [  ERROR] Login failed - unexpected server
 response!
2016-07-22 11:50:38,694 [     search] [   INFO] Failed to login to Pokemon Go. T
rying again.
2016-07-22 11:50:39,696 [   auth_ptc] [   INFO] PTC login for: Krysiz
2016-07-22 11:50:40,421 [   auth_ptc] [   INFO] PTC Login successful
2016-07-22 11:50:40,727 [     pgoapi] [  ERROR] Login failed - unexpected server
 response!
2016-07-22 11:50:40,729 [     search] [   INFO] Failed to login to Pokemon Go. T
rying again.
2016-07-22 11:50:41,732 [   auth_ptc] [   INFO] PTC login for: Krysiz
2016-07-22 11:50:42,509 [   auth_ptc] [   INFO] PTC Login successful
2016-07-22 11:50:42,736 [     pgoapi] [  ERROR] Login failed - unexpected server
 response!
2016-07-22 11:50:42,736 [     search] [   INFO] Failed to login to Pokemon Go. T
rying again.`

**Comment by HinoReo** [comment]
I am also getting these.

**Comment by Hoborific** [comment]
Consider the following: it's down

**Comment by HinoReo** [comment]
No it's not. I can login into game and via their site. Login server status also showing OK. Furthermore, the first login attempt is successful.

**Comment by Krys1s** [comment]
Please don't be a smart ass and read the original description.

I can close it out and restart and it connects fine. If it was down, that wouldn't happen. I believe there is a re-authentication that occurs after 30 minutes that seems to start failing.

**Comment by Hoborific** [comment]
PTC is down and there is pretty widespread threads popping up about it already.

**Comment by parrots** [comment]
FWIW here's my log running --debug

`2016-07-22 15:19:48,989 [connectionpool] [   INFO] Starting new HTTPS connection (1): sso.pokemon.com
2016-07-22 15:19:59,965 [connectionpool] [  DEBUG] "GET /sso/login?service=https%3A%2F%2Fsso.pokemon.com%2Fsso%2Foauth2.0%2FcallbackAuthorize HTTP/1.1" 200 103
2016-07-22 15:20:00,953 [connectionpool] [  DEBUG] "GET /sso/login?service=https%3A%2F%2Fsso.pokemon.com%2Fsso%2Foauth2.0%2FcallbackAuthorize HTTP/1.1" 200 103
2016-07-22 15:20:04,621 [connectionpool] [  DEBUG] "POST /sso/login?service=https%3A%2F%2Fsso.pokemon.com%2Fsso%2Foauth2.0%2FcallbackAuthorize HTTP/1.1" 302 None
2016-07-22 15:20:08,153 [connectionpool] [  DEBUG] "POST /sso/login?service=https%3A%2F%2Fsso.pokemon.com%2Fsso%2Foauth2.0%2FcallbackAuthorize HTTP/1.1" 302 0
2016-07-22 15:20:20,211 [connectionpool] [  DEBUG] "GET /sso/oauth2.0/callbackAuthorize?ticket=ST-676225-2Hb4la2GEC9FvBMzMFzI-sso.pokemon.com HTTP/1.1" 500 None
2016-07-22 15:20:27,530 [connectionpool] [  DEBUG] "POST /sso/oauth2.0/accessToken HTTP/1.1" 200 103
2016-07-22 15:20:27,531 [   auth_ptc] [   INFO] PTC Login successful
2016-07-22 15:20:27,534 [connectionpool] [   INFO] Starting new HTTPS connection (1): pgorelease.nianticlabs.com
2016-07-22 15:20:29,079 [connectionpool] [  DEBUG] "POST /plfe/rpc HTTP/1.1" 200 2
2016-07-22 15:20:29,083 [     pgoapi] [  ERROR] Login failed - unexpected server response!
2016-07-22 15:20:29,083 [     search] [   INFO] Failed to login to Pokemon Go. Trying again.
2016-07-22 15:20:30,091 [   auth_ptc] [   INFO] PTC login for: _usernameremoved_
2016-07-22 15:20:30,094 [connectionpool] [   INFO] Starting new HTTPS connection (1): sso.pokemon.com
2016-07-22 15:20:35,405 [connectionpool] [   INFO] Resetting dropped connection: sso.pokemon.com
2016-07-22 15:20:37,868 [connectionpool] [  DEBUG] "GET /503?locale=null HTTP/1.1" 301 0
2016-07-22 15:20:37,872 [connectionpool] [   INFO] Starting new HTTP connection (1): sso.pokemon.com
2016-07-22 15:20:41,704 [connectionpool] [  DEBUG] "GET /sso/login?service=https%3A%2F%2Fsso.pokemon.com%2Fsso%2Foauth2.0%2FcallbackAuthorize HTTP/1.1" 200 103`

**Comment by SimonTellsAll** [comment]
This also happens with google logins as well. As I am using a google account and eventually after a certain time it won't authenticate and goes through a loop of trying to authenticate and login.

**Comment by Krys1s** [comment]
If it was temporarily down why would a restart of the script authenticate correctly and function properly? You are self proclaimed more technically literate than the rest of us, so please enlighten.

I understand PTC is having issues NOW, this issue has been happening for the past 6 hours for me and after reproducing it about 10 times I opened the ticket.

**Comment by Hoborific** [comment]
You're right I misread the original post, you want something like #1292 for when PTC starts sending back incomplete data or empty objects.

I apologize

**Comment by Krys1s** [comment]
Awesome, didn't see the PR before, thanks!

**Comment by lrsuperman** [comment]
I just installed the newest developer build and received the exact same login failures.  I used my credentials on the website successfully.

**Comment by AyalaJuan** [comment]
Just came back - I was having the same problem.

**Comment by delrio1110** [comment]
same here

**Comment by Krys1s** [comment]
Just to clarify for people - PTC is completely broken right now so all logins are failing. The issue in this ticket was prior to that where PTC was working but the script authentication was hanging when it tried to re-authenticate mid session.

Symptom was an authentication failure loop that was resolved by restarting the script.

**Comment by Galastorm** [comment]
Have they found a way of detecting that these programs are being used?  I created a separate PTC account last night (though I logged in to the actual app on my phone to get past the initial setup steps) and used the master branch to log spawn points and times for my area.

Tonight, the burner account is coming back with login failures as is every other ptc account used on my phone.  All other devices/ptc accounts within my household (3 in all) are able to log in fine to the app.

It only seems to be a problem on my accounts...
